include/bout/region.hxx

```diff
@@ -938,4 +949,9 @@ unsigned int size(const Region<T> &region){
   return region.size();
 }
 
+// template<typename T>
+class RegionID {
```

Comment:
Please could you stick a docstring on this explaining its purpose? It probably also wants to be a `struct` if you're expecting to access `value`